EIPS/eip-6963.md

```diff
@@ -0,0 +1,118 @@
+
+Browser extensions are loaded in the web page and do not have a predictable pattern resulting in a race condition where the user is not in control to choose which wallet provider must be selected for exposing Ehtereum interface under the `window.ethereum` object.
+
+This results not only in a degraded user experience but also increases the barrier to entry for new browser extensions as users are forced to only install one browser extension at a time.
+
+Some browser extensions attempt to counteract this problem by delaying their injection to overwrite the same `window.ethereum` object which creates an unfair competition for wallet providers and lack of interoperability.
+
+In this proposal, we provide a solution that optimitizes for interoperability and enables fairer competition by lowering the barrier to entry for new wallet providers to improve the user experience on Ethereum.
+
+This is achieved by replacing the current window object `window.ethereum` with a new window object `window.evmproviders` that enables multiple browser extensions to be exposed for user selection in the web page.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Provider Info
+
+Each wallet provider will be exposed with the following interface `ProviderInfo` that will be used to display to the user:
+
+```typescript
+/**
+ * Represents the assets needed to display a wallet
+ */
+interface ProviderInfo {
+  uuid: string;
+  walletId: string;
+  name: string;
+  icon: string;
+}
+```
+
+The values in the `ProviderInfo` MUST be used as follows:
+
+- uuid - locally unique of the wallet provider (UUIDv4 compliant)
+- walletId - globally unique identifier of the walelt provider (eg. `my-wallet-extension` or `awesomewallet`)
+- name - human-readable name of the wallet provider (e.g. `MyWalletExtension` or `AwesomeWallet`)
+- icon - base64 encoded image (RFC-4648 complaint)
+
+### EVMProvider
+
+The wallet provider will also expose their own EIP-1193 provider interface in parallel with the provider info provided above in the following interface `EVMProvider`
+
+```typescript
+interface EVMProvider {
+  info: ProviderInfo;
+  ethereum: EIP1193Provider;
+}
+```
+
+The `EIP1193Provider` interface is documented at [EIP-1193](./eip-1193.md) and can be used to override the `window.ethereum` object once the user as explicitly selected it.
+
+### window.evmproviders
+
+A web app will be able to display multiple wallet providers that were injected by browser extensions by parsing the window object `window.evmproviders` which will include an array of wallet providers that follow the `EVMProvider` described above
```

Comment:
From my understanding, that falls under a phase two of this proposal (if at all):

> This EIP doesn't require supplanting window.ethereum, so it doesn't directly break existing applications. However, the recommended behavior of eventually supplanting window.ethereum would break existing applications that rely on it.

Even with a Proxy, the developer would need a list of injected wallets to make a decision on which they would want to load. Once this decision is made, a Proxy object wrapping `window.ethereum` could route requests to the intended provider. This feels like a nicer solution vs just overwriting `window.ethereum` once an EVM provider has been selected, but is addressing a slightly different problem in the provider selection chain.